lib/plugins/aws/package/compile/events/eventBridge/index.js

```diff
@@ -17,7 +17,9 @@ class AwsCompileEventBridgeEvents {
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'eventBridge', {
       type: 'object',
       properties: {
-        eventBus: { type: 'string', minLength: 1 },
+        eventBus: {
+          anyOf: [{ type: 'string', minLength: 1 }, { type: 'object' }],
```

Comment:
This could property could be an event bus name/arn (string) or a CloudFormation function (object)

Reply:
The change to the schema is coupled to this PR, it should only support an `object` (CloudFormation Func) off the back of this work. The CustomResource deployment pattern does not support them. To avoid updating the schema now I should remove the support for CloudFormation functions, which I would rather not do.

Reply:
@stuartforrest-infinity Ok,so if I understand clearly this is meant to open the door for CloudFormation intrinsic functions.

In that case we need to use dedicated definition -> https://github.com/serverless/serverless/blob/9399f2b89c8a841d1d7d96a22a8de640d8214479/lib/plugins/aws/provider/awsProvider.js#L261 (assuming we want to support all of them)

Also we need to validate this cleanly in Custom Resource case. So if we detect this notation being used we throw a meaningful error to the user, that this notation is not supported, but is supported with `shouldUseCloudFormation` option.

Reply:
We should be able to support all of them. Good point about validating for Custom Resources, I hadn't thought of that. Are there any utilities in the project already to check if a value is a CloudFormation intrinsic function? I've already rolled my own but don't want to duplicate that.

Reply:
>  Are there any utilities in the project already to check if a value is a CloudFormation intrinsic function?

It's good enough to complain if passed configuration is an object (which you can check with `_.isObject()`.

Object structure will be validated by config schema and we should not duplicate that with any inline logic

Reply:
@pgrzesik I think this was missed. Ideally instead of `type: 'object'`, we should clearly indicate we allow CF intrinsic function, and we have available definitions for that